### PR TITLE
Funding flow for Mainnet Controller account

### PR DESCRIPTION
**Pull request by JunichiSugiura**
This PR also includes
- fix for `ts-lint` CI action
- Style fix in `<InsufficientFunds />`

Screenshot 2024-07-06 at 18 04 12

## Summary by CodeRabbit

- **New Features**
  - **DelegateAccount Component**: Dependencies updated to improve performance and stability.
  - **StarknetProvider Component**: Refactored to handle different chains, enhancing compatibility.
  - **TransferEth Component**: Simplified logic by removing unused states.
  - **CartridgeAccount Class**: Introduced new methods and parameters for improved account management and deployment.

- **Bug Fixes**
  - **ErrorAlert Component**: Enhanced visual representation using a new rotating icon effect.

- **Documentation**
  - Added new components and methods to documentation for better developer understanding and usage.

- **Refactor**
  - **Workflows**: Updated GitHub Actions to latest versions for improved CI/CD processes.
  - **Various Components**: Refactored imports and internal logic to optimize performance and maintainability.

- **Chores**
  - **Dependencies**: Added new dependencies to improve functionality and compatibility.

### CODE REVIEW

**Comment by tarrencev** [comment]
Looking good. Can we tone down the warning. 

"Controller is in Alpha
We recommend limiting deposits to necessary assets."

File: examples/starknet-react-next/src/components/StarknetProvider.tsx [hunk 1]
```diff
@@ -64,12 +66,19 @@ const connectors = [
     ],
     {
       url,
-      rpc: process.env.NEXT_PUBLIC_RPC_SEPOLIA,
+      rpc: process.env.NEXT_PUBLIC_RPC_MAINNET,
       // paymaster: {
       //   caller: shortString.encodeShortString("ANY_CALLER"),
       // },
-      // theme: "dope-wars",
+      theme: "dope-wars",
```
Comment (JunichiSugiura) [inline examples/starknet-react-next/src/components/StarknetProvider.tsx:73]: TODO: Revert before merge

File: packages/keychain/src/components/DeploymentRequired.tsx [hunk 2]
```diff
@@ -77,7 +84,10 @@ export function DeploymentRequired({
     >
       <Content alignItems="center">
         {status === Status.COUNTERFACTUAL &&
-          account.chainId === constants.StarknetChainId.SN_SEPOLIA && (
+          [
+            constants.StarknetChainId.SN_SEPOLIA,
+            constants.StarknetChainId.SN_MAIN,
```
Comment (JunichiSugiura) [inline packages/keychain/src/components/DeploymentRequired.tsx:89]: oh, how can I set `deployHash` from `<Funding />` hmmm

File: packages/keychain/src/components/Funding.tsx [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+          >
+            copy address
+          </Button>
+        )}
+      </Footer>
+    </Container>
+  );
+}
+
+function Warning({
```
Comment (JunichiSugiura) [inline packages/keychain/src/components/Funding.tsx:229]: This comp could be re-usable.

File: packages/keychain/src/components/Funding.tsx [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+    </VStack>
+  );
+}
+
+function ExternalWalletProvider({ children }: PropsWithChildren) {
+  const { connectors } = useInjectedConnectors({});
+
+  return (
+    <StarknetConfig
+      chains={[mainnet]}
```
Comment (JunichiSugiura) [inline packages/keychain/src/components/Funding.tsx:265]: `<Funding />` page only support Mainnet for now

File: packages/keychain/src/components/Funding.tsx [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+        address: ETH_CONTRACT,
+        min: "10000000000000000",
+      },
+      ...prefunds,
+    ];
+    if (!target.length) return;
+    fetchTokneInfo();
+
+    async function fetchTokneInfo() {
+      const res = await fetch("https://mainnet-api.ekubo.org/tokens");
```
Comment (JunichiSugiura) [inline packages/keychain/src/components/Funding.tsx:297]: We should bootstrap this internally.

File: packages/keychain/src/components/Funding.tsx [hunk 6]
```diff
@@ -0,0 +1,390 @@
...
+
+type TokenInfo = {
+  name: string;
+  symbol: string;
+  decimals: number;
+  address: string;
+  logo: React.ReactNode;
+  min: string;
+  isFunded: boolean;
+};
```
Comment (JunichiSugiura) [inline packages/keychain/src/components/Funding.tsx:390]: - `TokenInfo` should have `balance: bigint` field.
- Should I convert `min` to bigint
- `isFunded` can be computed by `t.balance >= t.min`.

File: packages/keychain/src/hooks/connection.tsx [hunk 7]
```diff
@@ -63,14 +65,22 @@ export function ConnectionProvider({ children }: PropsWithChildren) {
   }, [context, parent]);
 
   useEffect(() => {
-    if (!isIframe()) {
+    if (isIframe()) {
```
Comment (JunichiSugiura) [inline packages/keychain/src/hooks/connection.tsx:68]: URL search params only exists on iframe right?

File: packages/keychain/src/components/Toaster.tsx [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import { Flex, UseToastOptions } from "@chakra-ui/react";
+
+export function Toaster({ title }: UseToastOptions) {
```
Comment (JunichiSugiura) [inline packages/keychain/src/components/Toaster.tsx:3]: Let's used this on `<InsufficientFunds />` also.

File: packages/keychain/src/components/Funding.tsx [hunk 9]
```diff
@@ -0,0 +1,390 @@
...
+              bg="solid.primary"
+              fontWeight="semibold"
+            >
+              {t.logo}
+              <Text w="full">
+                {t.address === ETH_CONTRACT
+                  ? formatEther(BigInt(t.min))
+                  : t.min}{" "}
+                {t.symbol}
+              </Text>
```
Comment (JunichiSugiura) [inline packages/keychain/src/components/Funding.tsx:188]: I should replace this with current balance

File: packages/keychain/src/components/Funding.tsx [hunk 10]
```diff
@@ -0,0 +1,390 @@
...
+              <Spacer />
+              {t.isFunded && <CheckIcon />}
```
Comment (JunichiSugiura) [inline packages/keychain/src/components/Funding.tsx:190]: and show min value here.

**Review by JunichiSugiura — COMMENTED** [review]
Some thoughts

File: packages/keychain/src/hooks/connection.tsx (hunk 7 above)
Comment (broody) [reply packages/keychain/src/hooks/connection.tsx:82]: Ah this broke `slot auth login` flow. It looks like we're passing the prefund data over URL so it needs this. Just going to remove this condition check. Policies, origins, etc can just be overridden connect.

**Pull request merged**